File: opensource/data/src/main/java/com/sos/scheduler/engine/data/folder/JobPath.java
package com.sos.scheduler.engine.data.folder;

import org.codehaus.jackson.annotate.JsonCreator;

public class JobPath extends TypedPath {
    public JobPath(AbsolutePath path) {
        super(FileBasedType.job, path);
    }

    @JsonCreator
    public static JobPath of(String absolutePath) {
        return new JobPath(new AbsolutePath(absolutePath));
    }
}

File: opensource/data/src/main/scala/com/sos/scheduler/engine/data/folder/JobPath.scala
package com.sos.scheduler.engine.data.folder

import com.fasterxml.jackson.annotation.JsonCreator

final case class JobPath(string: String)
extends TypedPath {

  assertIsAbsolute()

  def typ =
    FileBasedType.job
}


object JobPath {
  @JsonCreator def of(absolutePath: String): JobPath =
    new JobPath(absolutePath)

  def makeAbsolute(path: String) =
    new JobPath(AbsolutePath.makeAbsolute(path))
}
